ddtrace/contrib/pytest/plugin.py

```diff
@@ -116,29 +205,47 @@ def pytest_runtest_protocol(item, nextitem):
         yield
         return
 
-    with _CIVisibility._instance.tracer.trace(
+    if isinstance(item.parent.parent, pytest.Package):
+        if _extract_span(item.parent.parent) is None:
+            _start_test_module_span(item)
+
+    if _extract_span(item.parent) is None:
+        _start_test_suite_span(item)
+
+    with _CIVisibility._instance.tracer._start_span(
         ddtrace.config.pytest.operation_name,
         service=_CIVisibility._instance._service,
         resource=item.nodeid,
         span_type=SpanTypes.TEST,
+        activate=True,
     ) as span:
         span.set_tag_str(COMPONENT, "pytest")
-        span.set_tag(SPAN_KIND, KIND)
-        span.set_tag(test.FRAMEWORK, FRAMEWORK)
-        span.set_tag(test.NAME, item.name)
+        span.set_tag_str(SPAN_KIND, KIND)
+        span.set_tag_str(test.FRAMEWORK, FRAMEWORK)
+        span.set_tag_str(_EVENT_TYPE, SpanTypes.TEST)
+        span.set_tag_str(test.NAME, item.name)
+        test_session_span = _extract_span(item.session)
+        span.set_tag(_SESSION_ID, test_session_span.span_id)
+        test_suite_span = _extract_span(item.parent)
+        span.set_tag(_SUITE_ID, test_suite_span.span_id)
```

Comment:
it looks as though this tag ends up showing up in `meta` but not in the root of the span. It needs to be in the root

Reply:
Unfortunately with our current implementation of converting traces into events, we have to set this tag when creating these spans, but I've added logic into the CI Visibility-specific encoder to remove this when the corresponding event is being created.